Race in TestTwistedConnection.test_connection_initialization

```
 ____________ TestTwistedConnection.test_connection_initialization _____________
  
  self = <tests.unit.io.test_twistedreactor.TestTwistedConnection testMethod=test_connection_initialization>
  
      def test_connection_initialization(self):
          """
          Verify that __init__() works correctly.
          """
          self.mock_reactor_cft.assert_called_with(self.obj_ut.add_connection)
  >       self.mock_reactor_run.assert_called_with(installSignalHandlers=False)
  
  D:\a\python-driver\python-driver\tests\unit\io\test_twistedreactor.py:115: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  
  self = <MagicMock name='run' id='2480883044064'>, args = ()
  kwargs = {'installSignalHandlers': False}
  expected = 'run(installSignalHandlers=False)', actual = 'not called.'
  error_message = 'expected call not found.\nExpected: run(installSignalHandlers=False)\nActual: not called.'
  
      def assert_called_with(self, /, *args, **kwargs):
          """assert that the last call was made with the specified arguments.
      
          Raises an AssertionError if the args and keyword args passed in are
          different to the last call to the mock."""
          if self.call_args is None:
              expected = self._format_mock_call_signature(args, kwargs)
              actual = 'not called.'
              error_message = ('expected call not found.\nExpected: %s\nActual: %s'
                      % (expected, actual))
  >           raise AssertionError(error_message)
  E           AssertionError: expected call not found.
  E           Expected: run(installSignalHandlers=False)
  E           Actual: not called.
```

--- a/tests/unit/io/test_twistedreactor.py
+++ b/tests/unit/io/test_twistedreactor.py
@@ -98,6 +98,7 @@ class TestTwistedConnection(unittest.TestCase):
         self.reactor_cft_patcher = patch(
             'twisted.internet.reactor.callFromThread')
         self.reactor_run_patcher = patch('twisted.internet.reactor.run')
+        self.thread_patcher = patch('cassandra.io.twistedreactor.Thread')
         # Patch reactor.running to False so maybe_start() always enters
         # the branch that spawns the reactor thread. Without this, leaked
         # reactor state from prior tests can cause reactor.running to be
@@ -107,6 +108,9 @@ class TestTwistedConnection(unittest.TestCase):
             'twisted.internet.reactor.running', new=False)
         self.mock_reactor_cft = self.reactor_cft_patcher.start()
         self.mock_reactor_run = self.reactor_run_patcher.start()
+        self.mock_thread_class = self.thread_patcher.start()
+        self.mock_thread = self.mock_thread_class.return_value
+        self.mock_thread.is_alive.return_value = False
         self.reactor_running_patcher.start()
         self.obj_ut = twistedreactor.TwistedConnection(DefaultEndPoint('1.2.3.4'),
                                                        cql_version='3.0.1')
@@ -114,6 +118,7 @@ class TestTwistedConnection(unittest.TestCase):
     def tearDown(self):
         self.reactor_cft_patcher.stop()
         self.reactor_run_patcher.stop()
+        self.thread_patcher.stop()
         self.reactor_running_patcher.stop()
 
     def test_connection_initialization(self):
@@ -121,7 +126,12 @@ class TestTwistedConnection(unittest.TestCase):
         Verify that __init__() works correctly.
         """
         self.mock_reactor_cft.assert_called_with(self.obj_ut.add_connection)
-        self.mock_reactor_run.assert_called_with(installSignalHandlers=False)
+        self.mock_thread_class.assert_called_once_with(
+            target=self.mock_reactor_run,
+            name="cassandra_driver_twisted_event_loop",
+            kwargs={'installSignalHandlers': False})
+        self.assertIs(self.mock_thread.daemon, True)
+        self.mock_thread.start.assert_called_once_with()
 
     def test_client_connection_made(self):
         """

